### PR TITLE
Scala 2.11.7 using dev

### PR DESCRIPTION
**Pull request by ligadata-saleh**
a new pull request to dev branch instead of release 1.2 branch .

### CODE REVIEW

**Comment by ligadata-william** [comment]
I receive an error using the KVInit tool:
```
ERROR [main] - Failed to connect to Datastore. Reason:null. Message:scala.reflect.api.JavaUniverse.runtimeMirror(Ljava/lang/ClassLoader;)Lscala/reflect/api/JavaMirrors$JavaMirror;
Cause:
java.lang.NoSuchMethodError: scala.reflect.api.JavaUniverse.runtimeMirror(Ljava/lang/ClassLoader;)Lscala/reflect/api/JavaMirrors$JavaMirror;
	at com.ligadata.kamanja.samples.containers.V1000000.SputumCodes.getWithReflection(SputumCodes.scala:125)
	at com.ligadata.kamanja.samples.containers.V1000000.SputumCodes.get(SputumCodes.scala:94)
	at com.ligadata.tools.kvinit.KVInit$$anonfun$com$ligadata$tools$kvinit$KVInit$$buildContainerOrMessage$1$$anonfun$apply$9$$anonfun$17.apply(KVInit.scala:868)
	at com.ligadata.tools.kvinit.KVInit$$anonfun$com$ligadata$tools$kvinit$KVInit$$buildContainerOrMessage$1$$anonfun$apply$9$$anonfun$17.apply(KVInit.scala:865)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:245)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:245)
	at scala.collection.IndexedSeqOptimized$class.foreach(IndexedSeqOptimized.scala:33)
	at scala.collection.mutable.ArrayOps$ofRef.foreach(ArrayOps.scala:186)
	at scala.collection.TraversableLike$class.map(TraversableLike.scala:245)
	at scala.collection.mutable.ArrayOps$ofRef.map(ArrayOps.scala:186)
	at com.ligadata.tools.kvinit.KVInit$$anonfun$com$ligadata$tools$kvinit$KVInit$$buildContainerOrMessage$1$$anonfun$apply$9.apply(KVInit.scala:865)
	at com.ligadata.tools.kvinit.KVInit$$anonfun$com$ligadata$tools$kvinit$KVInit$$buildContainerOrMessage$1$$anonfun$apply$9.apply(KVInit.scala:826)
	at scala.collection.immutable.List.foreach(List.scala:381)
	at com.ligadata.tools.kvinit.KVInit$$anonfun$com$ligadata$tools$kvinit$KVInit$$buildContainerOrMessage$1.apply(KVInit.scala:826)
	at com.ligadata.tools.kvinit.KVInit$$anonfun$com$ligadata$tools$kvinit$KVInit$$buildContainerOrMessage$1.apply(KVInit.scala:819)
	at scala.collection.IndexedSeqOptimized$class.foreach(IndexedSeqOptimized.scala:33)
	at scala.collection.mutable.ArrayOps$ofRef.foreach(ArrayOps.scala:186)
	at com.ligadata.tools.kvinit.KVInit.com$ligadata$tools$kvinit$KVInit$$buildContainerOrMessage(KVInit.scala:819)
	at com.ligadata.tools.kvinit.KVInit$.main(KVInit.scala:202)
	at com.ligadata.tools.kvinit.KVInit.main(KVInit.scala)
```

**Comment by ligadata-william** [comment]
Looks like I need to switch my version of scala to 2.11. That'd probably help. :P

**Comment by ligadata-saleh** [comment]
also clean the ivy2 cache folder to get the correct jars for scala 2.11 , and install the same kafka version that is compatible with sacala 2.11

**Comment by ligadata-william** [comment]
Fair points. I'll do that. Thanks.

File: trunk/FileDataConsumer/build.sbt [hunk 1]
```diff
@@ -56,19 +57,20 @@ excludedJars in assembly <<= (fullClasspath in assembly) map { cp =>
...
-   // "io.spray" %%  "spray-json" % "1.2.5",
+   // "io.spray" %% "spray-can" % sprayVersion,
+   // "io.spray" %% "spray-routing" % sprayVersion,
+   // "io.spray" %% "spray-testkit" % sprayVersion,
+   // "io.spray" %% "spray-client" % sprayVersion,
+   // "io.spray" %%  "spray-json" % "1.3.2",
 
-   // "org.apache.kafka" % "kafka_2.10" % "0.8.1.1",
+   // "org.apache.kafka" %% "kafka" % "0.8.2.1",
     "org.apache.kafka" % "kafka_2.10" % "0.8.2.2",
```
Comment (ligadata-william) [inline trunk/FileDataConsumer/build.sbt]: This kafka dependency is showing kafka_2.10. I believe this should be kafka_2.11 if we're upgrading to scala 2.11.x libraries.

File: trunk/InputOutputAdapters/KafkaSimpleInputOutputAdapters/build.sbt [hunk 2]
```diff
@@ -11,7 +11,7 @@ libraryDependencies += "org.apache.logging.log4j" % "log4j-api" % "2.4.1"
 
 libraryDependencies += "org.apache.logging.log4j" % "log4j-core" % "2.4.1"
 
-libraryDependencies ++= Seq("org.apache.kafka" % "kafka_2.10" % "0.8.1.1"
+libraryDependencies ++= Seq("org.apache.kafka" %% "kafka" % "0.8.2.1"
```
Comment (ligadata-william) [inline trunk/InputOutputAdapters/KafkaSimpleInputOutputAdapters/build.sbt]: Here kafka version 0.8.2.1 is being referenced, but above, kafka 0.8.2.2 is referenced. They should be the same version.

File: trunk/FileDataConsumer/build.sbt (hunk 1 above)
Comment (ligadata-saleh) [inline trunk/FileDataConsumer/build.sbt]: yes , the micro change the commit instead for the line , 

"org.apache.kafka" % "kafka_2.10" % "0.8.2.2",
should be 
"org.apache.kafka" %% "kafka" % "0.8.2.1", 

i will change it

File: trunk/InputOutputAdapters/KafkaSimpleInputOutputAdapters/build.sbt (hunk 2 above)
Comment (ligadata-saleh) [inline trunk/InputOutputAdapters/KafkaSimpleInputOutputAdapters/build.sbt]: actually 0.8.2.1 is the correct version ,  all jars pointed to that version , i change the above to 0.8.2.1 as well

Comment (ligadata-krishna) [inline trunk/InputOutputAdapters/KafkaSimpleInputOutputAdapters/build.sbt]: I thought you are doing pattern match to replace 2.10 with 2.11.

Comment (ligadata-saleh) [inline trunk/InputOutputAdapters/KafkaSimpleInputOutputAdapters/build.sbt]: i did that , but some dependencies went missing after compiling the code , some jar versions is not supported in for scala 2.11 , i am checking the micros if there is anything that needs to be fixed

**Pull request merged**